controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -243,3 +256,47 @@ func (r *DSCInitializationReconciler) updateStatus(ctx context.Context, original
 
 	return saved, err
 }
+
+func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsci.DSCInitialization) error {
+	if instance.Spec.ServiceMesh.ManagementState == operatorv1.Managed {
+		serviceMeshInitializer := feature.NewFeaturesInitializer(&instance.Spec, servicemesh.ConfigureServiceMeshFeatures)
+
+		if err := serviceMeshInitializer.Prepare(); err != nil {
+			r.Log.Error(err, "failed configuring service mesh resources")
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed configuring service mesh resources")
+
+			return err
+		}
+
+		if err := serviceMeshInitializer.Apply(); err != nil {
+			r.Log.Error(err, "failed applying service mesh resources")
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying service mesh resources")
+
+			return err
+		}
+	}
```

Comment:
I agree... Although I'd lean towards calling `Prepare` inside `feature.NewFeaturesInitializer`.

Reply:
A bit of a background from where the names are coming:

- `Prepare` define features and process templates etc.
- `Apply` and `Delete` make actual operations on the cluster

If I understand it correctly you would like to have `Prepare` and `Delete` where the former does the cluster operations?

Reply:
Nope, I'm good with `Apply` and `Delete` as is.

My suggestion was that maybe `feature.NewFeaturesInitializer` should return an already `Prepare`d initializer? rather than to `Prepare` inside `Apply` and `Delete`?

I see no reason to defer the `Prepare` to the inside of `Apply` and `Delete` since these two are anyways called immediately after instantiating the initializer.